ext/pcntl/php_signal.c

```diff
@@ -48,18 +54,19 @@ Sigfunc *php_signal4(int signo, Sigfunc *func, int restart, int mask_all)
 		act.sa_flags |= SA_RESTART; /* SVR4, 4.3+BSD */
 #endif
 	}
-	if (zend_sigaction(signo, &act, &oact) < 0) {
-		return (void*)SIG_ERR;
+	if (sigaction(signo, &act, NULL) < 0) {
+		return FAILURE;
 	}
 
-#ifdef HAVE_STRUCT_SIGINFO_T
-	return oact.sa_sigaction;
-#else
-	return oact.sa_handler;
-#endif
+	/* Ensure this signal is not blocked */
+	sigemptyset(&sigset);
+	sigaddset(&sigset, signo);
+	php_sigprocmask(SIG_UNBLOCK, &sigset, NULL);
```

Comment:
Why is this necessary now?

Reply:
Previously this code called `zend_sigaction`, which (as a side effect) ensures that the signal is not blocked. Now that it is not using `zend_sigaction`, I added the code here to make sure that the behavior of `php_signal4` is not changed.

But maybe that extra side effect from `zend_sigaction` was not needed in the first place! Let me just look and see if that is true or not.

Reply:
Hmm. `pctnl_signal()` from `pcntl` uses `php_signal4`, and it doesn't do anything else to explicitly unblock signals. So if this code is removed, the behavior of `pcntl_signal()` will change subtly.

Does it matter? Is anyone actually relying on `pcntl_signal()` to unblock signals for them? I don't know, but I suggest that if a change to the behavior of `pcntl_signal` is wanted, it should be treated as a separate issue and not handled in this PR. This PR is just intended to simplify internals without changing user-visible behavior.